packages/salesforcedx-vscode-core/src/notifications/notificationService.ts

```diff
@@ -26,6 +29,30 @@ export class NotificationService {
     return NotificationService.instance;
   }
 
+  // Prefer these over directly calling the vscode.show* functions
+  // We can expand these to be facades that gather analytics of failures.
```

Comment:
Use these from the notificationService instead of invoking the functions from vscode.workspace.*